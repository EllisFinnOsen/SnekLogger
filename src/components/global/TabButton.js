import React from "react";
import { TouchableOpacity, Text } from "react-native";
import { useThemeColor } from "../../hooks/useThemeColor";
import { SIZES, FONT } from "../../constants/Theme";

// Dynamic style functions
const buttonStyle = (
  name,
  activeTab,
  activeColor,
  subtleTextColor,
  iconColor
) => ({
  paddingVertical: SIZES.small,
  paddingHorizontal: SIZES.large,
  backgroundColor: name === activeTab ? activeColor : subtleTextColor,
  borderRadius: SIZES.xSmall,
  marginLeft: 2,
  shadowColor: iconColor,
});

const buttonTextStyle = (name, activeTab, textColor, bgColor) => ({
  fontFamily: FONT.medium,
  fontSize: SIZES.medium,
  color: name === activeTab ? textColor : bgColor,
});

export default function TabButton({
  name,
  activeTab,
  onHandleSearchType,
  activeColor,
  subtleTextColor,
  iconColor,
  textColor,
  bgColor,
}) {
  return (
    <TouchableOpacity
      style={buttonStyle(
        name,
        activeTab,
        activeColor,
        subtleTextColor,
        iconColor
      )}
      onPress={onHandleSearchType}
    >
      <Text style={buttonTextStyle(name, activeTab, textColor, bgColor)}>
        {name}
      </Text>
    </TouchableOpacity>
  );
}
