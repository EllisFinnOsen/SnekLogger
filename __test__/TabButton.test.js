import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import TabButton from "@/components/global/TabButton"; // Adjust the import path as needed

describe("TabButton", () => {
  const defaultProps = {
    name: "Home",
    activeTab: "Home",
    onHandleSearchType: jest.fn(),
    activeColor: "blue",
    subtleTextColor: "gray",
    iconColor: "red",
    textColor: "white",
    bgColor: "black",
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders the button with the correct text", () => {
    const { getByText } = render(<TabButton {...defaultProps} />);
    expect(getByText("Home")).toBeTruthy();
  });

  it("calls onHandleSearchType when pressed", () => {
    const onPressMock = jest.fn();
    const props = { ...defaultProps, onHandleSearchType: onPressMock };
    const { getByText } = render(<TabButton {...props} />);
    // Since the Text is inside the TouchableOpacity, firing an event on the Text will bubble up.
    const buttonText = getByText("Home");
    fireEvent.press(buttonText);
    expect(onPressMock).toHaveBeenCalled();
  });

  it("applies active styles when name equals activeTab", () => {
    // In active state, the button background should be activeColor and the text color should be textColor.
    const props = { ...defaultProps, name: "Home", activeTab: "Home" };
    const { toJSON } = render(<TabButton {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("applies inactive styles when name does not equal activeTab", () => {
    // In inactive state, the button background should be subtleTextColor and the text color should be bgColor.
    const props = { ...defaultProps, name: "Home", activeTab: "Profile" };
    const { toJSON } = render(<TabButton {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
