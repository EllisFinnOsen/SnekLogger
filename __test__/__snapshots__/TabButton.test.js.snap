// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButton applies active styles when name equals activeTab 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "blue",
      "borderRadius": 10,
      "marginLeft": 2,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
      "shadowColor": "red",
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontFamily": "OutfitMed",
        "fontSize": 16,
      }
    }
  >
    Home
  </Text>
</View>
`;

exports[`TabButton applies inactive styles when name does not equal activeTab 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "gray",
      "borderRadius": 10,
      "marginLeft": 2,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
      "shadowColor": "red",
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontFamily": "OutfitMed",
        "fontSize": 16,
      }
    }
  >
    Home
  </Text>
</View>
`;
